pkg/agent/cniserver/secondary.go

```diff
@@ -56,6 +62,15 @@ func (pc *podConfigurator) ConfigureSriovSecondaryInterface(
 	containerConfig := buildContainerConfig(hostInterfaceName, containerID, podName, podNamespace,
 		containerNetNS, containerIface, result.IPs, 0)
 	pc.ifaceStore.AddInterface(containerConfig)
+	sriovInterface := &interfacestore.SRIOVInterfaceConfig{
+		InterfaceConf: containerConfig,
+		VFDeviceID:    podSriovVFDeviceID,
+		ResourceName:  resourceName,
+	}
+	pc.sriovIfaceStore.Save(filestore.AnyObjectWithUID{
```

Comment:
We should at least check and log the error. Should we roll back the interface creation or have some way to re-try saving the interface?

Reply:
Checked the returned error and added a log. I didn't find a proper way to roll back the interface creation. And I feel it's probably not necessary to do it only because of the SRIOV interface config store failure. (The worst case without proper SRIOV interface cleanup is that VF device will be back to host without the original name, and it can still be reused by CNI next time).
Retry might be doable, there are two possible store operation failures, one is file open failure, another is encode failure, I am not sure if it's worthy to retry for them. I feel we can log the error for now as what the NP controller did in a similar way. https://github.com/antrea-io/antrea/blob/main/pkg/agent/controller/networkpolicy/networkpolicy_controller.go#L311-L313.

Reply:
The original purpose was for VF name restoration. but when I did this change, I found a couple of issues after agent restart (without this change):

1. Pod's secondary nic IP won't be released in IPPool.
2. SecondaryNetwork controller didn't handle any existing Pod's update event anymore after restart.
3. SecondaryNetwork controller won't call removeInterfaces for the secondary nic because of empty storeInterfaces for any existing Pods before restart.

Reply:
1. This issue does not require restoring SR-IOV interfaces. As long as we restore cniCache (which does not require a file store), it can be solved.
2. And then the problem is VF cannot be renamed back to its original name? Any other problem?
3. Really? I think CNI del will release IPs of the Pod's all secondary interfaces (by calling ipamDel -> ipam.SecondaryNetworkDel).

Reply:
Hi @jianjuns 

I didn't notice `ipam.SecondaryNetworkDel` in CNI del. Based on the test I did, it seems the IP release is done by the `SecondaryNetworkRelease` inside of SecondaryNetwork's `removeInterfaces`: https://github.com/antrea-io/antrea/blob/main/pkg/agent/secondarynetwork/podwatch/controller.go#L267-L270.  So, for now, SecondaryNetwork's `removeInterfaces` will be responsible for VF rename and IP release.

I will check why SecondaryNetworkDel didn't work as expected.

Reply:
Hi @jianjuns the root cause for SecondaryNetworkDel failure is here https://github.com/antrea-io/antrea/blob/main/pkg/agent/cniserver/server.go#L220-L234, it seems it's never working as expected because it returns too early without configuring `cniConfig.secondaryNetworkIPAM = true`.

Reply:
Since we still want to support annotation deletion and recreation in this PR https://github.com/antrea-io/antrea/pull/7119, I suppose we still need a separate way to delete secondary interfaces?

Per my understand, the SecondaryNetworkDel would be called only when the primary interface is deleted. And the original purpose of SecondaryNetworkDel seems to cover the AntreaIPAM with other CNIs, like multus. https://github.com/antrea-io/antrea/pull/3529.